client/apps/edit/client.coffee

```diff
@@ -13,20 +13,94 @@ EditAdmin = require './components/admin/index.coffee'
 EditDisplay = require './components/display/index.coffee'
 SectionList = React.createFactory require './components/section_list/index.coffee'
 HeroSection = React.createFactory require './components/hero_section/index.coffee'
+async = require 'async'
 
 @init = ->
   article = new Article sd.ARTICLE
-  channel = new Channel sd.CURRENT_CHANNEL
-  new EditLayout el: $('#layout-content'), article: article, channel: channel
-  new EditHeader el: $('#edit-header'), article: article
-  new EditDisplay el: $('#edit-display'), article: article
-  new EditAdmin el: $('#edit-admin'), article: article, channel: channel
-  React.render(
-    SectionList(sections: article.sections)
-    $('#edit-sections')[0]
-  )
-  if article.get('hero_section') != null or channel.hasFeature 'hero'
+  convertSections article, =>
+    channel = new Channel sd.CURRENT_CHANNEL
+    new EditLayout el: $('#layout-content'), article: article, channel: channel
+    new EditHeader el: $('#edit-header'), article: article
+    new EditDisplay el: $('#edit-display'), article: article
+    new EditAdmin el: $('#edit-admin'), article: article, channel: channel
     React.render(
-      HeroSection(section: article.heroSection, hasSection: article.get('section_ids')?.length > 0)
-      $('#edit-hero-section')[0]
+      SectionList(sections: article.sections)
+      $('#edit-sections')[0]
     )
+    if article.get('hero_section') != null or channel.hasFeature 'hero'
+      React.render(
+        HeroSection(section: article.heroSection, hasSection: article.get('section_ids')?.length > 0)
+        $('#edit-hero-section')[0]
+      )
+
+convertSections = (article, callback) ->
+  async.parallel(
+    article.sections.map (section) ->
+      (cb) ->
+        if section.get('type') is 'image'
+          convertImages section, cb
+        else if section.get('type') is 'artworks'
+          convertArtworks section, cb
+        else if section.get('type') is 'image_set'
+          convertImageSet section, cb
+        else
+          cb()
+  , =>
+    callback()
+  )
+
+convertImageSet = (section, callback) ->
+  async.parallel(
+    section.get('images').map (image) -> (cb) ->
+      img = new Image()
+      img.src = image.image or image.url # image or artwork
+      img.onload = ->
+        image.width = img.width
+        image.height = img.height
+        image.artists = [image.artist] if image.type is 'artwork'
```

Comment:
yeah i was going to do the same conversion process for image sets but i think we said that the frontend doesn't support it yet, right?